actors/builtin/power/power_actor.go

```diff
@@ -215,11 +215,20 @@ func (a Actor) OnEpochTickEnd(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 	_, code := rt.Send(
 		builtin.RewardActorAddr,
 		builtin.MethodsReward.UpdateNetworkKPI,
-		&st.TotalRawBytePower,
+		&st.ThisEpochRawBytePower,
 		abi.NewTokenAmount(0),
 	)
 	builtin.RequireSuccess(rt, code, "failed to update network KPI with Reward Actor")
 
+	// update next epoch's power and pledge values
```

Comment:
There is a question of whether we should be doing this before we send raw byte power to reward. This seems right but I wouldn't be too surprised if I was mistaken here.

Reply:
I think we want it before the send. At the end of an epoch, we want to freeze the current power values, and then use them for computing the next epoch's block reward. Other calculations in terms of total power will then use the matching values for the block reward current during that epoch.

Describe this in a comment so that the ordering is transparent to a new reader.